styx-scheduler-service/src/main/java/com/spotify/styx/state/EventFeeder.java

```diff
@@ -0,0 +1,89 @@
+/*-
+ * -\-\-
+ * Spotify Styx Common
+ * --
+ * Copyright (C) 2016 - 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.state;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.spotify.styx.util.IsClosed;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Single threaded asynchronous event consumer queue. It requires a {@link Consumer}
+ * implementation to act upon the queued events of type {@link T}.
+ */
+public class EventFeeder<T> implements Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(EventFeeder.class);
+  private static final int SHUTDOWN_GRACE_PERIOD_SECONDS = 5;
+
+  private final Consumer<T> eventConsumer;
+  private final ThreadPoolExecutor executor;
+
+  public EventFeeder(Consumer<T> eventConsumer) {
+    this.eventConsumer = Objects.requireNonNull(eventConsumer);
+    this.executor = new ThreadPoolExecutor(1, 1, 0L,
+        TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>());
+  }
+
+  void enqueue(T event) throws IsClosed {
```

Comment:
shall we call this `newEvent` or `onEvent` to not to disclose `queue` which is rather a internal design of this class?

Reply:
Or `feed` if we called the class Feeder.

But consistency would also be achieved with `EventQueue` + `enqueue()`, or `EventConsumerExecutor` + `execute()`.

My overarching point is _it's not very useful to have a Styx-specific abstraction over a vanilla ExecutorService API here_. It just makes it necessary for users to read doc (realistically, read code) to understand what added value Feeder brings over an ExecutorService.

Reply:
A very good comment, I almost went the same direction. So, let's get rid of feeder completely.